packages/yourloops/test/unit/components/footer/footer.test.tsx

```diff
@@ -75,53 +75,24 @@ describe('Footer', () => {
     }
   })
 
-  function checkLinkHref(linkId: string, expectedUrl: string) {
-    const link = document.getElementById(linkId) as HTMLLinkElement
-    expect(link.href).toBe(expectedUrl)
+  function getLanguageSelector(): HTMLElement {
+    return document.getElementById('footer-language-box')
   }
 
-  it('should render', async () => {
+  it('should render language selector on routes requiring it', async () => {
     await mountComponent()
-    const component = document.getElementById('footer-links-container')
-    expect(component).not.toBeNull()
-  })
-
-  it('should render language selector when user is not logged in', async () => {
-    await mountComponent()
-    const languageSelector = document.getElementById('footer-language-box')
-    expect(languageSelector).not.toBeNull()
+    ROUTES_REQUIRING_LANGUAGE_SELECTOR.forEach((route: string) => {
+      history.replace(route)
+      expect(getLanguageSelector()).not.toBeNull()
+    })
   })
 
-  it('should not render language selector when user is logged in', async () => {
-    await mountComponent({} as User)
+  it('should not render language selector when user is logged in on home page', async () => {
+    history.replace('/home')
```

Comment:
To be coherent, instead on testing that the language selector is not present on the `/home` route, I would do like the test above (iterate over all the routes not supposed to have the language selector).

The only issue with your current solution (ut vs it), is that if another route without languague selector is added, this test needs to be updated in order to add the route. A solution to that is to have an array with all the routes (or all the routes without language selector) that is actually used in the code and from which your test is based.